src/DirectForceApplication.cpp

```diff
@@ -0,0 +1,46 @@
+#include <gazebo_usv/DirectForceApplication.hpp>
+
+#include <string>
+
+#include <gazebo/physics/Link.hh>
+#include <gazebo/physics/Model.hh>
+#include <gazebo/physics/World.hh>
+
+#include <gazebo_usv/Actuators.hpp>
+
+using namespace gazebo_usv;
+
+DirectForceApplication::~DirectForceApplication() {
+    delete mActuator;
+}
+
+void DirectForceApplication::load(gazebo::physics::ModelPtr model, gazebo::physics::LinkPtr link, gazebo::transport::NodePtr node) {
+    mActuator = new Actuators(model, node);
+    mLinkId = mActuator->addLink(link);
+
+    // Initialize communication node and subscribe to gazebo topic
+    std::string topicName = model->GetName() + "/" + link->GetName() + "/directional_force";
```

Comment:
maybe a name that would be more specific ... like `gazebo_usv_force` ?

Reply:
I think I can namespace the node with `gazebo_usv` in the initialize method, so the publisher's topic will also be namespace as well

Reply:
I would personally feel that the philosophy behind topics would be to put the gazebo_usv more to the end (to clearly attach the topic to the link). No ?

Reply:
At first, I was thinking about assuring that every topic (published or subscribed) in `gazebo_usv` is prefixed with the package name (an old custom with ros).

But it would interfere with the thrusters plugin, so I'll just keep it as it is and use the `gazebo_usv_force` in the very end, as you said.

Reply:
> At first, I was thinking about assuring that every topic (published or subscribed) in gazebo_usv is prefixed with the package name (an old custom with ros).

LOL

So ... ROS went from

"Topics are great because the single source of truth for a quantity is just named "/pose", super intuitive"

To: "well, it's a mess, let's prefix every single topic with the name of the node that generates it"

Tbh, I haven't worked extensively with topics, I'm good either way.